typescript/sdk/src/deploy/verify/ContractVerifier.ts

```diff
@@ -108,38 +108,58 @@ export class ContractVerifier {
         method: 'GET',
       });
     } else {
-      response = await fetch(url.toString(), {
+      const init: RequestInit = {
         method: 'POST',
         headers: { 'Content-Type': 'application/x-www-form-urlencoded' },
         body: params,
-      });
+      };
+      response = await fetch(url.toString(), init);
+    }
+    let responseJson;
+    try {
+      const responseTextString = await response.text();
+      responseJson = JSON.parse(responseTextString);
+    } catch (error) {
+      verificationLogger.trace(
+        {
+          failure: response.statusText,
+          status: response.status,
+          chain,
+          apiUrl,
+          family,
+        },
+        'Failed to parse response from explorer.',
+      );
+      throw new Error(
+        `Failed to parse response from explorer (${apiUrl}, ${chain}): ${
+          response.statusText ?? 'UNKNOWN STATUS TEXT'
+        } (${response.status ?? 'UNKNOWN STATUS'})`,
```

Comment:
probably don't need the nullish check. Maybe better to use `||`

according to the [spec](https://fetch.spec.whatwg.org/#response-class), this can return an empty byte, but never undefined.

Reply:
nice– updated